app.rb

```diff
@@ -2,123 +2,166 @@
 require './student'
 require './book'
 require './teacher'
+require './save_data'
 
 class App
-  def show_books(books)
-    if books.empty?
+  attr_reader :book, :person, :rental
+
+  include DataSaver
+
+  def initialize
+    @book = show_books
+    @person = show_people
+    @rental = show_rentals(@book, @person)
+    @student_data = get_data('people')
+    @teach_data = get_data('people')
+    @book_data = get_data('books')
+    @rental_data = get_data('rental')
+  end
+
+  def list_books
+    if @book.empty?
       puts 'No books found in library'
     else
-      books.each do |book|
-        puts "title: #{book.title}, Auther: #{book.author}"
+      book.each do |item|
+        puts "Title: #{item.title}, Author: #{item.author}"
       end
     end
   end
 
-  def show_people(people)
-    if people.empty?
+  def list_people
+    if @person.empty?
       puts 'No person found'
     else
-      people.each do |person|
-        puts "Name: #{person.name}, Age: #{person.age},ID: #{person.id}"
+      person.each do |item|
+        puts "[#{item.class}] - Name: #{item.name}, Age: #{item.age}, ID: #{item.id}"
```

Comment:
- [ ] Great work with the refactoring. However, I noticed that when I create people and list them, they have a certain `ID` but when I exit the app and start it again, their IDs change when I input the number for listing people. The implication of this is that whenever I create a rental with the ID of the person, the rental can't be retrieved. Kindly fix this bug also. 👍